### DISCUSSION TITLE
Choosing blockCV approaches

### DISCUSSION THREAD

**Discussion by sdowd20** [sdmTMB/sdmTMB#401] [Q&A] [answered]
Hi all,
I am using sdmTMB to generate density models for humpback whales in the Gulf of Maine. As with many cetaceans, the data on humpback whales in this area is sparse. The data I am using are humpback whale density estimates from a combination of many different aerial and shipboard surveys. 

I am going to use cross-validation with blockCV to find the "best" model for humpback whale density. I am debating whether I should create blocks based on the spatial or clustering methods in this data-limited scenario. If I used the clustering methods, I would create the clusters based on geographic space. Does anyone have any helpful guidance on how to decide which cross-validation approach to use? Are these two methods relatively similar if clustering considers space as well? 

Thanks!
Sally

**Comment by ericward-noaa** [comment, marked as answer]
Great question -- and this is a tough one! I have not been able to develop good guidance around how to design the blocks in a general way, using blockCV or other approaches. Estimates, and cross-validation scores are pretty sensitive to the choices of how you set the blocks up. It's frustrating, because in an ideal world you'd set up the blocks so that those in the training set are somewhat independent of those in the test set. If the distances between the training and test blocks << spatial range, then they won't be independent, but if the distance is larger than the spatial range, then the model will be predicting beyond the range of the data. Other than trying a few different permutations, I don't have other suggestions unfortunately.

**Reply by sdowd20** [reply]
Hi @ericward-noaa thank you for your response! 

I will try out a few different permutations. Right now, I have blocks defined as 2x the spatial autocorrelation distance through spatial blocking. I think I will try a kmeans approach as well.
